e2e-tests/demand-backup-cloud/run

```diff
@@ -68,6 +68,71 @@ run_recovery_from_source() {
 	compare_mysql_cmd "select-1" "SELECT * from myApp.myApp;" "-h $cluster-pxc-2.$cluster-pxc -uroot -proot_password"
 }
 
+# If backup upload was started and failed for some reaason the cloud storage should be cleaned up during second try
+delete_backup_pod() {
+	local backup_name=$1
+	local max_retry=120
+
+	echo "Waiting for ${backup_pod} to become Running"
+	set -o xtrace
+	# retry=0
+	sleep 1
+	kubectl wait --for=jsonpath='{.status.phase}'=Running pod --selector=percona.com/backup-job-name=xb-${backup_name} --timeout=120s
+	# until kubectl get pods --selector=percona.com/backup-job-name=xb-${backup_name} -o jsonpath='{.items[].status.phase}' 2>/dev/null | grep -q -i 'Running'; do
+	# 	sleep 1
+	# 	echo -n .
+	# 	let retry+=1
+
+	# 	if [[ $retry -ge $max_retry ]]; then
+	# 		echo max retry count $retry reached. something went wrong with operator or kubernetes cluster
+	# 		exit 1
+	# 	fi
+	# done
+
+	backup_pod=$(kubectl get pods --selector=percona.com/backup-job-name=xb-${backup_name} -o jsonpath='{.items[].metadata.name}')
+
+	log echo "Deleting pod/$backup_pod during SST upload"
+	kubectl logs -f ${backup_pod} | while IFS= read -r line; do
+		if [[ $line =~ \.ibd\. ]]; then
+			# give time to upload some data
+			kubectl delete pod --force $backup_pod
+			break
+		fi
+	done
+
+	# sleep till new backup pod starts
+	sleep 3
+}
+
+check_cloud_storage_cleanup() {
+	local backup_name=$1
+
+	if [ $(kubectl get events --field-selector involvedObject.kind=Job,involvedObject.name=xb-${backup_name} | grep -c "Created pod") -ne '2' ]; then
+		echo "There should be 2 pods started by job. First backup finished too quick"
+		exit 1
+	fi
+	local backup_pod=$(kubectl get pods --selector=percona.com/backup-job-name=xb-${backup_name} -o jsonpath='{.items[].metadata.name}')
+	if kubectl logs $backup_pod | egrep -i 'xbcloud delete|Delete completed.'; then
+		echo "Bucket cleanup was successful"
+	else
+		echo "Bucket cleanup was not performed!"
+		kubectl logs $backup_pod
+		exit 1
+	fi
+}
+
+run_backup_with_delete() {
+	local cluster=$1
+	local backup1=$2
```

Comment:
maybe `local backup=$2` would be better?

Reply:
Updated to `backup_name` to be consistent in all the test.